x/auction/abci.go

```diff
@@ -7,17 +7,16 @@ import (
 // EndBlocker runs at the end of every block.
 func EndBlocker(ctx sdk.Context, k Keeper) {
 
-	// get an iterator of expired auctions
-	expiredAuctions := k.GetQueueIterator(ctx, EndTime(ctx.BlockHeight()))
-	defer expiredAuctions.Close()
-
-	// loop through and close them - distribute funds, delete from store (and queue)
-	for ; expiredAuctions.Valid(); expiredAuctions.Next() {
-
-		auctionID := k.DecodeAuctionID(ctx, expiredAuctions.Value())
-		err := k.CloseAuction(ctx, auctionID)
+	var expiredAuctions []uint64
```

Comment:
Move this logic to a keeper function called `CloseExpiredAuctions` and just call it here. Follows the cosmos-sdk pattern of not putting logic in `abci.go` and wrapping it in a keeper method.

Reply:
good idea